backend/test/app.e2e-spec.ts

```diff
@@ -15,6 +15,10 @@ describe('AppController (e2e)', () => {
     await app.init();
   });
 
+  afterAll(async () => {
+    await app.close();
```

Comment:
These tests fail as the there is no mongodb here to reach. While the approach of providing a mockmodule like in the other tests will not work here there is a simple solution that works here:


```js
 const dataSource = await memoryMongoDataSource()

    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    })
    .overrideProvider(DataSource)
      .useValue(dataSource)
      .compile();

    app = moduleFixture.createNestApplication();
    await app.init();

```
    
With the helper function being:

```js
     export async function memoryMongoDataSource() {
  mongod = await MongoMemoryServer.create();
  const mongoUri = await mongod.getUri();
  return new DataSource({ type: 'mongodb', url: mongoUri });
};
```